src/services/providers/provider-sources/load-balance-provider.ts

```diff
@@ -31,46 +31,62 @@ export class LoadBalanceProviderSource implements IProviderSource {
 function loadBalance(transports_: readonly Transport[], config: LoadBalanceProviderSourceConfig = {}): Transport {
   const { minSuccessRate = 0.05, minSamples = 3, maxAttempts, samplesTtl = '30m' } = config;
 
-  const rpcMetrics: MethodMetrics[] = transports_.map(() => ({}));
-
   return ({ chain, timeout, ...rest }) => {
-    const transports = transports_.map((t) => t({ chain, timeout, ...rest }));
+    const transports = transports_
+      .map((t) => t({ chain, timeout, ...rest }))
+      .map((transport) => new TransportInstance(transport, { samplesTtl }));
     return createTransport({
       key: 'load-balance',
       name: 'Load Balancing',
+      type: 'load-balance',
       async request({ method, params }): Promise<any> {
-        let availableTransports = transports.map((transport, index) => ({ transport, index }));
+        const availableTransports: Record<string, TransportInstance> = Object.fromEntries(
```

Comment:
We will use a `Record` so that it's easier to delete transports by id once they have failed